src/components/ColorSwatch.js

```diff
@@ -1,80 +1,132 @@
-import theme from 'prism-react-renderer/themes/github';
+import PropTypes from 'prop-types';
+import {
+  SprkStack,
+  SprkStackItem,
+  SprkDivider,
+  SprkHeading,
+  SprkText,
+} from '@sparkdesignsystem/spark-react';
+import classnames from 'classnames';
+import useColorData from '../hooks/use-color-data';
 
-const ColorSwatch = ({ name, description, hexValue, rgbValue, sassVar, swatchHeight }) => (
-  <div className="sprk-c-Card sprk-o-Stack">
-    <div className="sprk-o-Stack__item">
-      <div
-        style={{
-          minHeight: swatchHeight || '140px',
-          backgroundColor: rgbValue
-        }}
-        className="sprk-c-Card__media"></div>
-    </div>
+const nameFormatter = (name) => {
+  return name
+    .replace(/-/g, ' ')
+    .replace('sprk', '')
+    .split(' ')
+    .map((word) => word.substring(0, 1).toUpperCase() + word.substring(1))
+    .join(' ');
+};
+
+const ColorSwatch = (props) => {
+  const colorData = useColorData();
 
-    <div className="
-      sprk-o-Stack__item
-      sprk-c-Card__content
-      sprk-o-Stack
-      sprk-o-Stack--large
-      "
-    >
-      <h3 className="sprk-b-TypeDisplayFive sprk-o-Stack__item">
-        {name}
-      </h3>
+  const { variableName, colorName, isCompact, hasOutline } = props;
 
-      <p className="sprk-b-TypeBodyTwo sprk-o-Stack__item">
-        {description}
-      </p>
+  // return a single object containing data for one color
+  const colorNode = colorData.filter((item) => {
+    return item.node.context.name === variableName;
+  })[0];
 
-      <div className="sprk-o-Stack sprk-o-Stack--medium">
-        {hexValue && (
-          <div className="sprk-o-Stack__item sprk-o-Stack sprk-o-Stack--medium">
-            <h4 className="sprk-b-TypeDisplaySeven sprk-o-Stack__item">
-              Hex Value
-            </h4>
+  const calculatedName = nameFormatter(colorNode.node.context.name);
+  const rgbValue = colorNode.node.context.value;
+  const { description } = colorNode.node;
+  const hexValue = colorNode.node.type;
+  const sassVar = `$${colorNode.node.context.name}`;
 
-            <InlineCode
-              className="css"
-              theme={theme}
-              additionalPreClasses="sprk-u-pas sprk-u-man sprk-o-Stack__item"
+  if (isCompact) {
+    return (
+      <div
+        className={classnames(
+          'docs-c-ColorSwatch__swatch docs-c-ColorSwatch__swatch--compact sprk-u-mrs',
+          { 'docs-c-ColorSwatch__swatch--hasOutline': hasOutline },
+        )}
+        style={{
+          backgroundColor: hexValue,
+        }}
+      />
+    );
+  }
+  return (
+    <>
+      <SprkStack additionalClasses="docs-c-ColorSwatch" splitAt="large">
+        <SprkStackItem additionalClasses="sprk-o-Stack__item--two-fifths@l">
+          <SprkStack splitAt="tiny">
+            <SprkStackItem
+              className={classnames(
+                'docs-c-ColorSwatch__swatch sprk-u-mrs',
+                // eslint-disable-next-line prettier/prettier
```

Comment:
why did you have to disable prettier on this line?

Reply:
Prettier kept reformatting it and breaking the `classnames` syntax.